spec/components/topic_query_spec.rb

```diff
@@ -233,6 +233,34 @@
     end
   end
 
+  context 'before and after filter' do
+    before do
+      Fabricate(:topic, title: '10 day old topic', created_at: 10.days.ago)
+      Fabricate(:topic, title: '15 day old topic', created_at: 15.days.ago)
+      Fabricate(:topic, title: '30 day old topic', created_at: 30.days.ago)
+    end
+
+    it "filters before date topics correctly" do
+      expect(TopicQuery.new(user).list_latest.topics.size).to eq(3)
+      expect(TopicQuery.new(user, before: '20').list_latest.topics.size).to eq(1)
+      expect(TopicQuery.new(user, before: '20').list_latest.topics.first.title).to eq('30 day old topic')
```

Comment:
We can store the results of the query in a variable here so we don't query the database for something we already have.